observer.go

```diff
@@ -27,6 +27,11 @@ type PeerObservation struct {
 	Peer    Server
 }
 
+// HeartbeatObservation is sent when a node fails to heartbeat with the leader
+type HeartbeatObservation struct {
```

Comment:
I wonder if instead we should call this `FailedHeartbeatObservation` or if we should instead always send an observation for every heartbeat and include a `Failed` boolean

Reply:
Yeah, I think `FailedHeartbeatObservation` would be a better choice. It seems like there could be a bit of overhead if we observed every heartbeat, especially if the heartbeat interval is low. Also I'd imagine that there aren't many cases where developers want to observe successful heartbeats.